### ISSUE TITLE
Incorrect colors in typescript code

### ISSUE THREAD

**Issue by cfsmp3** [Microsoft/TypeScript-Sublime-Plugin#680]
(originally posted [here](https://forum.sublimetext.com/t/incorrect-colors-in-typescript-code/39401), but they sent me your way)

Doesn't happen to all the code, but it happens to this specific bit, see screenshot.

![sublime_missingcolors](https://user-images.githubusercontent.com/5949913/47235139-f55a0600-d38c-11e8-9703-9b645825c958.png)

What is most interesting and I can't explain is that if I comment out that console.log line, everything works as expected.

![sublime_colorsok](https://user-images.githubusercontent.com/5949913/47235156-0571e580-d38d-11e8-921e-20d0acf57730.png)

**Comment by DanielRosenwasser** [comment]
Are you using the most up-to-date version of the plugin, and if not, can you post a minimal reproduction of the code that has this error?

**Comment by cfsmp3** [comment]
@DanielRosenwasser Yes, just checked, 3.1.1, and this simple thing triggers it:

![sublime_minimal](https://user-images.githubusercontent.com/5949913/47246210-6f9c8180-d3b1-11e8-9fc8-ebd103935b58.png)


Again commenting out that first console.log() line is enough to to get it to work as expected.

**Comment by DanielRosenwasser** [comment]
Can you actually post code? Screenshots aren't copy/paste-able 😉

**Comment by cfsmp3** [comment]
Sure. This should do:

Version with wrong colors:
```
function dummy()
{
    return Promise.all([])
        .then((result) => // Array of arrays (one array per user,each listing the token
     {
        const send_all_messages_promises = [];
        console.log('Result of device list', result);
        const usedTokens = {};
        for (const userdevicelist of result) {
            for (const registrationTokenObj of userdevicelist) {
                const registrationToken = registrationTokenObj.registrationToken;
                const deviceRef = registrationTokenObj.reference;
                console.log("Token", registrationToken);
            }
        }
    });
}
```

Version with correct colors:

`function dummy () 
{
	return Promise.all ([])
        .then((result) =>
     {
        const send_all_messages_promises = [];
        console.log('Result of device list', result);
        const usedTokens = {};
        for (const userdevicelist of result) {
            for (const registrationTokenObj of userdevicelist) {
                const registrationToken = registrationTokenObj.registrationToken;
                const deviceRef = registrationTokenObj.reference;
                console.log("Token", registrationToken);
            }
        }
    });
}`

**Comment by DanielRosenwasser** [comment]
This issue was moved to Microsoft/TypeScript-TmLanguage#678

**Comment by DanielRosenwasser** [comment]
Released [3.1.5](https://github.com/Microsoft/TypeScript-Sublime-Plugin/releases/tag/3.1.5) with updated syntax highlighting.
